### PR TITLE
licensing: enforce branding

### LINKED ISSUE #14028: RFC 167: Enforce branding
Part of [RFC 167](https://docs.google.com/document/d/1XozQ4JINJqirdaG-XqGtboT2-PlIXPyBn6EwV7Q3pWI/edit).

### PR DESCRIPTION
**Pull request by unknwon**
This PR:

1. Removes custom branding for OSS
1. Returns branding info when the license permits
1. Produces a site alert for site admins when they try to custom branding but the license does not permit

~~Stack on #14943,~~ fixes #14028

### CODE REVIEW

File: enterprise/cmd/frontend/internal/licensing/init/init.go [hunk 1]
```diff
@@ -27,6 +28,40 @@ func Init(ctx context.Context, enterpriseServices *enterprise.Services) error {
 	// proper product name.
 	graphqlbackend.GetProductNameWithBrand = licensing.ProductNameWithBrand
 
+	conf.Branding = func() *schema.Branding {
+		if !licensing.EnforceTiers ||
+			licensing.CheckFeature(licensing.FeatureBranding) == nil {
```
Comment (keegancsmith) [inline enterprise/cmd/frontend/internal/licensing/init/init.go:33]: `conf.Branding` is called a lot (on ever HTTP page load), but it seems `licensing.CheckFeatures` does a bit of work. Are there concerns here about it being slow?

**Review by keegancsmith — APPROVED** [review]
Approving, but this feels weird to inject custom behaviour by setting a variable in the `conf` package. Is there not a nicer way this can be done. EG the callers of `conf.Branding` probably should be calling something else / have information passed into them.

File: enterprise/cmd/frontend/internal/licensing/init/init.go (hunk 1 above)
Comment (unknwon) [reply enterprise/cmd/frontend/internal/licensing/init/init.go:33]: I wouldn't say it is slow but if `conf.Branding` is called a lot, definitely valid use case to use `conf.Watch`. I'll refactor.

**Comment by unknwon** [comment]
Merging as-is, happy to address any post-merge comments!

**Pull request merged**